qutip/core/qobjevo.py

```diff
@@ -376,33 +216,14 @@ def f2_t(t, args):
         Apply the function f to every Qobj. f(Qobj) -> Qobj
         Return a modified QobjEvo and let the original one untouched
 
-    apply_decorator(decorator, *args, str_mod=None,
-                    inplace_np=False, **kw_args):
-        Apply the decorator to each function of the ops.
-        The *args and **kw_args are passed to the decorator.
-        new_coeff_function = decorator(coeff_function, *args, **kw_args)
-        str_mod : list of 2 elements
-            replace the string : str_mod[0] + original_string + str_mod[1]
-            *exemple: str_mod = ["exp(",")"]
-        inplace_np:
-            Change the numpy array instead of applying the decorator to the
-            function reading the array. Some decorators create incorrect array.
-            Transformations f'(t) = f(g(t)) create a missmatch between the
-            array and the associated time list.
-
     tidyup(atol=1e-12)
         Removes small elements from quantum object.
 
     compress():
         Merge ops which are based on the same quantum object and coeff type.
 
-    compile(code=False, matched=False, dense=False, omp=0):
+    compile():
         Create the associated cython object for faster usage.
-        code: return the code generated for compilation of the strings.
-        matched: the compiled object use sparse matrix with matching indices.
-                    (experimental, no real advantage)
-        dense: the compiled object use dense matrix.
-        omp: (int) number of thread: the compiled object use spmvpy_openmp.
```

Comment:
Compilation is now automatic, lot of test can be simplified.
But all test passed locally..., not sure why, they should not as you are saying...
I will clean the tests today.

Reply:
That's odd if they passed - I would have expected them to fail because there are lots of tests which feed `omp=2` and similar kwargs to `QobjEvo.compile`, and with these changes `QobjEvo` no longer knows about them.  It's certainly not a problem if they're passing, though.